sdk/typescript/src/client/cryptography.ts

```diff
@@ -0,0 +1,358 @@
+import {
+	create_dkg_centralized_output,
+	create_imported_dwallet_centralized_step,
+	create_sign_centralized_output,
+	decrypt_user_share,
+	encrypt_secret_share,
+	generate_secp_cg_keypair_from_seed,
+	network_dkg_public_output_to_protocol_pp,
+} from '@dwallet-network/dwallet-mpc-wasm';
+import { bcs } from '@mysten/sui/bcs';
+import { decodeSuiPrivateKey } from '@mysten/sui/cryptography';
+import { Secp256k1Keypair } from '@mysten/sui/keypairs/secp256k1';
+import sha3 from 'js-sha3';
+
+import { IkaClient } from './ika-client';
+import { DWallet } from './types';
+import { UserShareEncrytionKeys } from './user-share-encryption-keys';
+import { encodeToASCII, u64ToBytesBigEndian } from './utils';
+
+/**
+ * Prepared data for the second round of Distributed Key Generation (DKG).
+ * Contains all cryptographic outputs needed to complete the DKG process.
+ */
+export type PreparedSecondRound = {
```

Comment:
I don't like this struct. It mixes up two different things, and that's a high-risk of causing programming mistakes later on:
- the user (centralized party) output, inlucding its *private output: the secret key share*.
- the public output
- the message (which is the only part that somehow matches the struct name)

I much rather have it separated into `CentralizedPartyDKGPrivateOutput`, `CentralizedPartyDKGPublicOutput` and `CentralizedPartyDKGMessage`.



However, because you also hold here the secret key share un-encrypted, this structure isn't the input to the second round at all.